runtime/src/bpf_helper.cpp

```diff
@@ -64,20 +66,92 @@ long bpftime_strncmp(const char *s1, uint64_t s1_sz, const char *s2)
 	return strncmp(s1, s2, s1_sz);
 }
 
+static sigjmp_buf jump_buffer;
+
+// used for sigsetjmp to avoid faulting in probe_write_user & probe_read_user
+void segv_handler(int signum)
+{
+    siglongjmp(jump_buffer, 1);
+}
+
+
 uint64_t bpftime_probe_read(uint64_t dst, uint64_t size, uint64_t ptr, uint64_t,
 			    uint64_t)
 {
-	memcpy((void *)(uintptr_t)dst, (void *)(uintptr_t)ptr,
+	if(size<=0) {
+		return -EINVAL;
+	}
+	struct sigaction sa, old_sa;
+	uint64_t ret = 0;
+	
+	sa.sa_handler = segv_handler; // set signal handler
+	sigemptyset(&sa.sa_mask); // clear signal set
+	sa.sa_flags = 0;
+
+	if (sigaction(SIGSEGV, &sa, &old_sa) == -1) {
+		perror("sigaction");
+		return -EFAULT;
+	}
+	
+	if (sigsetjmp(jump_buffer, 1) == 0) {
+        // to avoid faulting for src & dst (which is NULL)	
+		memcpy((void *)(uintptr_t)dst, (void *)(uintptr_t)ptr,
 	       (size_t)(uint32_t)(size));
-	return 0;
+		spdlog::debug("probe_read: dst={}, src={}, len={}", dst, ptr, size);
+    } else {
+		spdlog::error("probe_read: failed to read from src={}", ptr);
```

Comment:
Please use macros to log, so we can control log level at compile time. For example, SPDLOG_ERROR(...)

Reply:
@yunwei37 suggested we use SPDLOG_TRACE here, I modify it by SPDLOG_TRACE. Thanks!

Reply:
No, for error here, you should use SPDLOG_ERROR.

For debug purposes, change to SPDLOG_DEBUG or SPDLOG_TRACE

Reply:
I got it, thanks.